### ISSUE TITLE
Workbench: error-handling if Flask application crashes

### ISSUE THREAD

**Issue by davemfish** [natcap/invest#1610]
If the Flask application crashes, the Workbench will not function very well. The UI will be usable, but new model tabs won't load, no validation can happen, etc. It's probably a good idea for electron to catch this when it happens and either try to restart the Flask app, or inform the user that they should close the Workbench and re-open it.

**Comment by dcdenu4** [comment]
I agree with Emily and James mostly from this [Slack thread](https://stanford-natcap.slack.com/archives/CEEGFC7K4/p1722979291072689). It seems like there are maybe two or three separate decisions here.

1) Best way to restart the Flask server proactively
This could be for updating languages, like Emily mentioned, when we know and have control over how to deal with the Flask server update. I agree here that asking the user to restart the application is a reasonable solution given the desktop nature of the app. As long as we are careful about how many settings / items would require a restart.

2) Should the Workbench be responsible for server endpoint exceptions, including Flask crashes?
Depending on complexity and effort I think it would be worthwhile to have some error handling or health monitor to attempt to gracefully let the user know when something has gone wrong. The solution might not be able to handle all instances, like James mentioned with OS-level process management or segfaults, but I think the Workbench UI should be responsible for updating it's connection status.

     I think a fair pattern would be for any endpoint in `ui_server.py` to return a `0|1` "exit code", indicating success or failure. I don't know the implementation details though, so this might not make sense.

3) How should the Workbench handle an unexpected Flask / endpoint crash?
I think "locking" the Workbench and asking the user to restart the application is a reasonable first solution here. If we already have code for restarting the Flask server, we could default to that with a message indicating what is happening and then have an option in a pop up window to manually restart the application too.

**Comment by davemfish** [comment]
I think based on the Slack conversation we would like to prompt the user to close & re-launch the Workbench (as opposed to try re-launching the flask process). We have similar functionality for when the user wants to change the language setting.

So the Workbench could listen for the exit of the python child process and take some action from there.

**Comment by davemfish** [comment]
We also have the `ErrorBoundary` component which displays after exceptions from the React tree. Could be worth considering if we want to re-use that in this case as well.
